tests/foreman/ui_airgun/test_contentcredential.py

```diff
@@ -24,712 +24,864 @@
+        product = entities.Product(gpg_key=gpg_key, organization=module_org).create()
+        repo = entities.Repository(product=product).create()
+        values = session.contentcredential.read(name)
+        assert values['details']['name'] == name
+        assert values['details']['content_type'] == 'GPG Key'
+        # transform string for comparison
+        transformed_string = gpg_content.replace('\n', ' ')
+        transformed_string = transformed_string.replace('  ', ' ')
+        assert values['details']['content'] == transformed_string
+        assert values['details']['products'] == '1'
+        assert values['details']['repos'] == '1'
+        assert values['products']['table'][0]['Name'] == product.name
+        assert values['repositories']['table'][0]['Name'] == repo.name
+        # Update gpg key with new name
+        session.contentcredential.update(
+            name,
+            {'details.name': new_name},
+        )
+        assert session.contentcredential.search(new_name)[0]['Name'] == new_name
+        # Delete gpg key
+        session.contentcredential.delete(new_name)
+        assert not session.contentcredential.search(new_name)
+
+
+@tier2
+def test_positive_add_empty_product(session, module_org, gpg_content):
+    """Create gpg key with valid name and valid gpg key then associate
+    it with empty (no repos) custom product
+
+    :id: e18ae9f5-43d9-4049-92ca-1eafaca05096
+
+    :expectedresults: gpg key is associated with product
+
+    :CaseLevel: Integration
+    """
+    prod_name = gen_string('alpha')
+    gpg_key = entities.GPGKey(
+        content=gpg_content,
+        organization=module_org,
+    ).create()
+    with session:
+        session.product.create(
+            {'name': prod_name, 'gpg_key': gpg_key.name})
+        values = session.contentcredential.read(gpg_key.name)
+        assert len(values['products']['table']) == 1
+        assert values['products']['table'][0]['Name'] == prod_name
+        assert values['products']['table'][0]['Used as'] == 'GPG Key'
+
+
+@tier2
+def test_positive_add_product_with_repo(session, module_org, gpg_content):
+    """Create gpg key with valid name and valid gpg key then associate it
+    with custom product that has one repository
+
+    :id: 7514b33a-da75-43bd-a84b-5a805c84511d
+
+    :expectedresults: gpg key is associated with product as well as with
+        the repository
+
+    :CaseLevel: Integration
+    """
+    name = gen_string('alpha')
+    gpg_key = entities.GPGKey(
+        content=gpg_content,
+        name=name,
+        organization=module_org,
+    ).create()
+    # Creates new product and associate GPGKey with it
+    product = entities.Product(
+        gpg_key=gpg_key,
+        organization=module_org,
+    ).create()
+    # Creates new repository without GPGKey
+    repo = entities.Repository(
+        url=FAKE_1_YUM_REPO,
+        product=product,
+    ).create()
+    with session:
```

Comment:
Think there a divergence between the test requirement and the implementation, as per doc we should:

setup phase:

1- create a  product 
2- create a repo in the product
3 - create a pgp_key  (the order of creating the pgp_key is not important at stage)

test phase:

4- in the UI via details/edit view associate the pgp_key